src/test/groovy/org/gradle/android/RelocationTest.groovy

```diff
@@ -1,135 +1,375 @@
 package org.gradle.android
 
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.TaskOutcome
+import org.gradle.util.GradleVersion
+import org.gradle.util.VersionNumber
 import spock.lang.Unroll
 
 import static org.gradle.testkit.runner.TaskOutcome.FROM_CACHE
+import static org.gradle.testkit.runner.TaskOutcome.NO_SOURCE
 import static org.gradle.testkit.runner.TaskOutcome.SUCCESS
+import static org.gradle.testkit.runner.TaskOutcome.UP_TO_DATE
 
 class RelocationTest extends AbstractTest {
 
     @Unroll
     def "simple Android app is relocatable with #gradleVersion and Android plugin #androidVersion"() {
+        assert gradleVersion instanceof GradleVersion
+        assert androidVersion instanceof VersionNumber
+
         def originalDir = temporaryFolder.newFolder()
-        new SimpleAndroidApp(originalDir, cacheDir, androidVersion.toString()).writeProject()
+        new SimpleAndroidApp(originalDir, cacheDir, androidVersion).writeProject()
 
         def relocatedDir = temporaryFolder.newFolder()
-        new SimpleAndroidApp(relocatedDir, cacheDir, androidVersion.toString()).writeProject()
+        new SimpleAndroidApp(relocatedDir, cacheDir, androidVersion).writeProject()
 
-        def gradleVerString = gradleVersion.version
+        def expectedResults = EXPECTED_RESULTS[androidVersion]
 
-        when:
-        def result = withGradleVersion(gradleVerString)
+        println expectedResults.describe(gradleVersion)
```

Comment:
Leftover println?

Reply:
This is intentionally here, comes in handy every time something breaks on Travis.